tests/FSharp.Data.GraphQL.Tests/OperationExecutionMiddlewareTests.fs

```diff
@@ -0,0 +1,86 @@
+module FSharp.Data.GraphQL.Tests.OperationExecutionMiddlewareTests
+
+open Xunit
+open FSharp.Data.GraphQL
+open FSharp.Data.GraphQL.Types
+open FSharp.Data.GraphQL.Parser
+open FSharp.Data.GraphQL.Execution
+open FSharp.Data.GraphQL.Ast
+
+#nowarn "40"
+
+type TestSubject = {
+    a: string
+    b: string
+    c: string
+}
+
+let data =
+    { a = "Apple"
+      b = "Banana"
+      c = "Cookie" }
+
+let DataType =
+    Define.Object<TestSubject>(
+        "Data",
+        fieldsFn = fun () ->
+        [
+            Define.Field("a", String, resolve = fun _ dt -> dt.a)
+            Define.Field("b", String, resolve = fun _ dt -> dt.b)
+            Define.Field("c", String, resolve = fun _ dt -> dt.c)
+        ])
+let Query =
+    Define.Object<TestSubject>(
+        "Query",
+        fieldsFn = fun () -> [ Define.Field("testData", DataType, (fun _ _ -> data)) ] )
+
+let schema = Schema(Query)
+
+let ast = parse """{
+        testData {
+            a
+            b
+            c
+        }
+    }"""
+
+let operationMiddleware (ctx : ExecutionContext) (next : ExecutionContext -> AsyncVal<GQLResponse>) =
```

Comment:
To test the middleware interface, I made a simple middleware that takes out one of the fields of the schema on the execution plan ("c" field).